crates/sqlexec/src/session.rs

```diff
@@ -231,10 +232,20 @@ impl Session {
 
     /// Create a physical plan for a given datafusion logical plan.
     pub async fn create_physical_plan(
-        &self,
+        &mut self,
         plan: DfLogicalPlan,
     ) -> Result<Arc<dyn ExecutionPlan>> {
         let state = self.ctx.init_exec();
+        if let DfLogicalPlan::Extension(extension) = &plan {
+            match extension.node.name() {
+                "CreateTable" => {
+                    let create_table = extension.node.as_any().downcast_ref::<CreateTable>();
+                    return self.create_table(create_table.unwrap().clone()).await;
+                }
+                _ => {}
+            }
+        }
+
```

Comment:
we want to handle the extensions using our session context, not datafusions, so we need to handle those cases here.